### PR TITLE
optional levenshtein edit distance 

### PR DESCRIPTION
**Pull request by cxfree**
@iandotkelly Wondering if you'd be interested in accepting this. This adds the ability to optionally use the Levenshtein edit distance to deduce licenses in LICENSE files.

### CODE REVIEW

**Comment by iandotkelly** [comment]
@cxfree - Thank you!  I am very happy for you to include this - but I might make one or two observations.

The main thing I might comment is that (in the license-find module) the defaultLicenses array and the patterns array do seem to overlap in their intention. But I can see the difficulty of merging the two approaches - if something says BSD it could be the 2 or 3 clause license.  

Also if you enable the edit distance matching, wouldn't you get hit from the simple regular expression matching and the edit distance?

I'm just wondering whether we need some way of merging those two approaches and removing duplicate results.  What do you think?

File: lib/edit-distance-match.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+var editDistance = require('fast-levenshtein').get,
+	assert = require('assert');
+
+
+function editDistanceMatch(text, licenseNames, spdxLicenses, take) {
+	assert(licenseNames.length > 0, 'licenseNames must be non-empty');
+	assert(Object.keys(spdxLicenses).length >= licenseNames.length,
+				 'spdxLicenses must be at least as long as licenseNames');
+
+	if (!take) {
```
Comment (iandotkelly) [inline lib/edit-distance-match.js]: This is more simply written as:

```js
take = take || 500;
```

**Comment by iandotkelly** [comment]
I am very inclined to accept this and build upon it.  Thank you so much for doing the work.

I presume this only works if the license is defined in its own file - if its appended to the bottom of a readme for example this wouldn't work - because the other content in the file would increase the edit distance.  When I considered doing this - I wondered whether I needed to compare the license to a segment of the file of approximately the size of the license - then scan over the file looking for a sufficiently low edit distance.  This is a lot more work to do however.

**Comment by cxfree** [comment]
Ah yes, good points - I'll have to think a bit more about how I can improve this.

**Comment by iandotkelly** [comment]
I'll re-open if you update, or if I decide to merge and fix the issues we discussed above

**Pull request closed without merging**